runelite-client/src/main/java/net/runelite/client/plugins/pyramidplunder/PyramidPlunderOverlay.java

```diff
@@ -0,0 +1,168 @@
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.pyramidplunder;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Shape;
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import net.runelite.api.GameObject;
+import net.runelite.api.ObjectComposition;
+import net.runelite.api.Point;
+import net.runelite.api.Varbits;
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import static net.runelite.client.plugins.pyramidplunder.PyramidPlunderObjects.GRAND_GOLD_CHEST_CLOSED_ID;
+import static net.runelite.client.plugins.pyramidplunder.PyramidPlunderObjects.GRAND_GOLD_CHEST_ID;
+import static net.runelite.client.plugins.pyramidplunder.PyramidPlunderObjects.SARCOPHAGUS_CLOSED_ID;
+import static net.runelite.client.plugins.pyramidplunder.PyramidPlunderObjects.SARCOPHAGUS_ID;
+import static net.runelite.client.plugins.pyramidplunder.PyramidPlunderObjects.SPEARTRAP_ID;
+import static net.runelite.client.plugins.pyramidplunder.PyramidPlunderObjects.TOMB_DOOR_CLOSED_ID;
+import static net.runelite.client.plugins.pyramidplunder.PyramidPlunderObjects.TOMB_DOOR_WALL_IDS;
+import static net.runelite.client.plugins.pyramidplunder.PyramidPlunderObjects.URN_CLOSED_IDS;
+import static net.runelite.client.plugins.pyramidplunder.PyramidPlunderObjects.URN_IDS;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayUtil;
+
+class PyramidPlunderOverlay extends Overlay
+{
+	private static final int MAX_DISTANCE = 2350;
+
+	private final Client client;
+	private final PyramidPlunderPlugin plugin;
+	private final PyramidPlunderConfig config;
+
+	@Inject
+	private PyramidPlunderOverlay(Client client, PyramidPlunderPlugin plugin, PyramidPlunderConfig config)
+	{
+		super(plugin);
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_SCENE);
+		this.client = client;
+		this.plugin = plugin;
+		this.config = config;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		Widget ppWidget = client.getWidget(WidgetInfo.PYRAMID_PLUNDER_DATA);
+		if (ppWidget == null)
+		{
+			return null;
+		}
+
+		ppWidget.setHidden(config.hideTimer());
+
+		// Highlight convex hulls of urns, chests, and sarcophagus
+		int currentFloor = client.getVar(Varbits.PYRAMID_PLUNDER_ROOM);
+		for (GameObject object : plugin.getObjectsToHighlight())
```

Comment:
These need distance checks probably. It highlights the objects in all of the rooms.